src/toolchain/web-guide.md

```diff
@@ -0,0 +1,135 @@
+
+- Web support with `gdext` is experimental and should be understood as such before proceeding.
+
+---
+
+
+## Installing Pre-Requisites
+
+- Install a nightly build of `rustc`, the `wasm32-unknown-emscripten` target for `rustc`, and `rust-src`.
+Assuming that Rust was installed with rustup, this is quite simple.
+
+  ```sh
+  rustup toolchain install nightly
+  rustup component add rust-src --toolchain nightly
+  rustup target add wasm32-unknown-emscripten --toolchain nightly
+  ```
+
+- Install Emscripten
+  - The simplest way to achieve this is to install `emsdk` from the git repo. Recommended to use version 3.1.39 for now[^1]
+
+  ```sh
+  git clone https://github.com/emscripten-core/emsdk.git
+  cd emsdk
+  ./emsdk install 3.1.39
+  ./emsdk activate 3.1.39
+  source ./emsdk.sh     (or ./emsdk.bat on windows)
+  ```
+
+  - It would also be HIGHLY recommended to follow the instructions in the terminal to add `emcc`[^2] to your `PATH`.
+    If not, it is necessary to manually `source` the `emsdk.sh` file in every new terminal prior to compilation.
+    This is platform-specific.
+
+[^1]: Note: Due to a bug with `emscripten`, the maximum version of `emcc`[^2] that can one compile `Godot` with is `3.1.39`.  `gdext` itself should be able to support the latest version of `emcc`, however, it may be a safer bet to stick to version `3.1.39`.
+[^2]: `emcc` is the name of Emscripten's compiler
+
+---
+
+
+## Project Configuration
+
+- Enable the `experimental-wasm` feature on `gdext` in the `Cargo.toml` file.
+  It is also strongly recommended to enable the `lazy-function-tables` feature to avoid an extremely long compile time with release builds.
+  It is likely to due a bug, but is still being investigated.  Edit the line to something like the following:
+
+  ```toml
+  godot = { git = "https://github.com/godot-rust/gdext", branch = "master", features = ["experimental-wasm", "lazy-function-tables"] }
+  ```
+
+- At the same level as your `Cargo.toml`, create a `.config` directory and inside of that,
+  create a `config.toml` file (if one doesn't already exist). The contents of this file should be as follows
+
+  ```toml
+  [target.wasm32-unknown-emscripten]
+  rustflags = [
+      "-C", "link-args=-sSIDE_MODULE=2",
+      "-C", "link-args=-sUSE_PTHREADS=1",
+      "-C", "target-feature=+atomics,+bulk-memory,+mutable-globals",
+      "-Zlink-native-libraries=no"
+  ]
+  ```
+
+- Edit the project's `.gdextension` to include the web path.
+  This is probably at `godot/EXTENSIONNAME.gdextension`.
+  The format will be similar to the following
+
+  ```gdextension
+  web.debug.wasm32 = "res://../rust/target/wasm32-unknown-emscripten/debug/EXTENSIONNAME.wasm"
+  web.release.wasm32 = "res://../rust/target/wasm32-unknown-emscripten/release/EXTENSIONNAME.wasm"
+  ```
+
+---
+
+
+## Compile the Project
+
+- Verify `emcc` is in the `PATH`. This can be as simple as doing the following:
+
+  ```sh
+  emcc --version
+  ```
+
+- Compile the code.
```

Comment:
Being specific just because it is also particularly useful to compile to your own target as well, so you can use the editor! (Maybe this can be mentioned somewhere? As a note like "If you `cargo clean`, don't forget to compile to your own target first (`cargo build`) so you can still use the editor" or something similar)

```suggestion
- Compile the code to WebAssembly (Emscripten).
```